src/solver/conv_winoRxS.cpp

```diff
@@ -413,115 +413,213 @@ ConvBinWinoRxS<Winodata, Winofilter>::Search(const ConvolutionContext& ctx,
-
-        if(GUI_predicted <= 0.1)
+            return -1.0;
+
+        // perf tables for the following kernel binaries:
+        // {stride1.f32, ostride2.f32, dstride2.f32} : {stride1.f16, ostride2.f16, dstride2.f16}
+        // should be updated in case of any changes in kernels configurations or implementaion
+
+        // F(2,3)
+        constexpr auto const_2_3 =
+            std::array<std::array<size_t, 3>, 2>{{{23652, 18349, 35391}, {28561, 58446, 40618}}};
+        constexpr auto fe_2_3 =
+            std::array<std::array<size_t, 3>, 2>{{{648, 690, 664}, {872, 737, 800}}};
+        constexpr auto ph_2_3 =
+            std::array<std::array<size_t, 3>, 2>{{{308, 326, 316}, {327, 337, 335}}};
+        constexpr auto be_2_3 =
+            std::array<std::array<size_t, 3>, 2>{{{2039, 2409, 1873}, {1852, 2533, 1593}}};
+
+        // F(3,2)
+        constexpr auto const_3_2 =
+            std::array<std::array<size_t, 3>, 2>{{{37329, 38554, 39701}, {41417, 49740, 57820}}};
+        constexpr auto fe_3_2 =
+            std::array<std::array<size_t, 3>, 2>{{{950, 616, 550}, {799, 779, 573}}};
+        constexpr auto ph_3_2 =
+            std::array<std::array<size_t, 3>, 2>{{{295, 315, 308}, {309, 329, 325}}};
+        constexpr auto be_3_2 =
+            std::array<std::array<size_t, 3>, 2>{{{6295, 9900, 7201}, {4941, 4927, 5111}}};
+
+        const size_t fp_idx     = is_fp16 ? 1 : 0;
+        const size_t stride_idx = ostride2 ? 1 : dstride2 ? 2 : 0 /*stride1*/;
+
+        const auto const_cost = (IS2X3 ? const_2_3 : const_3_2)[fp_idx][stride_idx];
+        const auto fe_cost    = (IS2X3 ? fe_2_3 : fe_3_2)[fp_idx][stride_idx];
+        const auto ph_cost    = (IS2X3 ? ph_2_3 : ph_3_2)[fp_idx][stride_idx];
+        const auto be_cost    = (IS2X3 ? be_2_3 : be_3_2)[fp_idx][stride_idx];
+
+        const auto S_loops  = Sg / S_factor;
+        const auto R_loops  = Rg / TILE_R;
+        const auto fe_calls = n_works_per_CU * S_loops * R_loops;
+        const auto be_calls = n_works_per_CU * (dstride2 ? 2 : 1);
+        const auto phases   = fe_calls * (Cg / C_factor) * (!single_mode && !dstride2 ? 2 : 1);
+
+        const auto cycles_predicted = static_cast<double>(const_cost + fe_cost * fe_calls +
+                                                          ph_cost * phases + be_cost * be_calls) /
+                                      1e6;
+
+        if(cycles_predicted <= 0.1)
             return -1.0; // Unreliable, too small work to do for the shader.
 
-        const auto N_MACS_PER_CU_PER_CLOCK = static_cast<size_t>(64) * 32 / DATATYPE_BITS;
-        const auto WTI_predicted           = direct_convolution_macs /
-                                   static_cast<double>(N_MACS_PER_CU_PER_CLOCK) /
-                                   static_cast<double>(n_groups) / GUI_predicted; // similar to BW
+        const size_t macs_per_cu_per_clock = 64 * (is_fp16 ? 2 : 1);
```

Comment:
Only true for gfx10 and gfx906 and earlier. However the fix should consider usage of the predicted wti.

Reply:
I can add more precise calculation based on GPU architecture, but I need some sort of documentation here and we have to decide what would be in a default case.
As far as I know, for gfx11 it should be multiplied by 2.